examples/ipam/main.tf

```diff
@@ -0,0 +1,28 @@
+module "vpc" {
+  source = "aws-ia/vpc/awscc"
```

Comment:
Fails to initialize. should be 
source  = "../.."

Reply:
No it shouldn't, because it leads to warnings in the Terraform Registry like below. We should update the repo-template so this is less confusing. Was planning to discuss this in the team meeting tomorrow. [Reference](https://www.terraform.io/docs/modules/create.html?_ga=2.84254169.1550907931.1648073135-1900038773.1641930758#standard-module-structure)

![image](https://user-images.githubusercontent.com/11078689/159803887-103f9f28-7837-4922-b043-e2cdb8565fee.png)

Reply:
It needs to be updated manually for testing changes locally, but you currently can't specify variables or conditional logic for the module's source or version.

Reply:
In this circumstance I agree with @tbulding . we leave example module references as `"../.."` to allow your examples dirs to test changes made to the module

open to discuss - there is a point of confusion for users because they often view examples as a something that you should be able to literally copy / paste. when you reference via "../.." that fails. ive considered making our examples something like this:

```hcl
module "vpc" {
  source = "../.."
  #  <LINK TO EXPLANATION WHY WE HAVE ../.. IN EXAMPLE>
  #  source = "aws-ia/vpc/awscc"
}
```

Reply:
With relative paths, the example is only usable as-is if called with the module code where defined, such as if the end user has cloned the repo, which is a big expectation imo. The primary users of examples are more likely to be less experienced, so it's best to keep the user experience as simple as possible. The trade-off is that module contributors have to override the source path in their clones to use the examples for testing, but trading a little DX for simpler UX is worth it imo.

Reply:
I have thought about that a lot. What I've considered is fully separating testing and examples. However, thats a pretty big burden burden on devs because:
1. now you have to maintain 2 sets of tests
2. since we still want to ensure examples/ entries work we have to test those

im not certain its worth it when you fully consider  that a comment may be "good enough" UX

Admittedly, the comment is not there currently [(except for ipam module)](https://github.com/aws-ia/terraform-aws-ipam/blob/2ddcf076bc4d8200d05b88e1ee73e5556561497f/examples/multiple_scopes/main.tf#L25) but we could consider adopting it